package/rdspgbadger.py

```diff
@@ -127,15 +131,26 @@ def write_log(client, dbinstance_id, filename, logfilename):
                     raise
         with open(filename, "a") as logfile:
             if 'LogFileData' in response:
-                logfile.write(response["LogFileData"])
+                downloaded_lines = response["LogFileData"].count("\n")
+                if response["AdditionalDataPending"] and downloaded_lines < max_number_of_lines:
+                    if downloaded_lines == 0:
+                        raise Exception("Not a single line was downloaded in last portion!")
+                    max_number_of_lines = max(downloaded_lines - 10, 1)
```

Comment:
I don't understand the `-10` here, why `10` specifically?

Reply:
Pretty arbitrary, see PR description.